src/pages/system-checks/index.js

```diff
@@ -0,0 +1,141 @@
+import { useState, useContext, useEffect } from 'react'
+import { Box } from '@mui/material'
+import toast from 'react-hot-toast'
+import Table from 'src/components/Shared/Table'
+import WindowToolbar from 'src/components/Shared/WindowToolbar'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { useWindowDimensions } from 'src/lib/useWindowDimensions'
+import { SystemRepository } from 'src/repositories/SystemRepository'
+import { DataSets } from 'src/resources/DataSets'
+import { CommonContext } from 'src/providers/CommonContext'
+import useResourceParams from 'src/hooks/useResourceParams'
+
+const SystemChecks = () => {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { height } = useWindowDimensions()
+  const { getAllKvsByDataset } = useContext(CommonContext)
+
+  //states
+  const [checkStore, setCheckStore] = useState([])
+  const [data, setData] = useState([])
+
+  const { labels: _labels, access } = useResourceParams({
+    datasetId: ResourceIds.SystemChecks
+  })
+
+  const handleSubmit = () => {
+    postChecks()
+  }
+
+  function getAllSystems() {
+    getAllKvsByDataset({
+      _dataset: DataSets.SYSTEM_CHECKS,
+      callback: setCheckStore
+    })
+  }
+
+  useEffect(() => {
+    if (checkStore) {
+      const resCheckedSystems = getRequest({
+        extension: SystemRepository.SystemChecks.qry,
+        parameters: `_scope=1`
+      })
+
+      Promise.all([resCheckedSystems]).then(([checkedSystems]) => {
+        const mergedSYCheckedList = checkStore.map(x => {
```

Comment:
is check store used to do anything other than mapping here? if no it is better not to define it as state as i guess

Reply:
checkStore state used here since it's required to store data returned from getAllKvsByDataset